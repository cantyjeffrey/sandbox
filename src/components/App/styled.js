import styled from 'styled-components';

export const Root = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 1rem;
  border: 1px solid red;
`;

export default {};
