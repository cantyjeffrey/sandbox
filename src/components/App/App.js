import React from 'react';
import { Root } from './styled';

const App = () => (
  <Root>
    <h1>Application</h1>
  </Root>
);

export default App;
