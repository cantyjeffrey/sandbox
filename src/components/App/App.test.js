import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import App from './App';

describe('<App />', () => {
  it('should match existing snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a headline with correct copy', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.text()).toEqual('Application');
  });

  it('renders with correct styles', () => {
    const wrapper = renderer.create(<App />).toJSON();
    expect(wrapper).toHaveStyleRule('border', '1px solid red');
  });
});
