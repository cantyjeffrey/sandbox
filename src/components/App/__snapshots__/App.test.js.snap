// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match existing snapshot 1`] = `
<Root>
  <h1>
    Application
  </h1>
</Root>
`;
